File: src/wheel/llm/llm_factory.py
from src.common import Factory

from .llm_engine import LLMEngine
from .llm_models import LLMConfig
from .imp.deepseek_engine import DeepSeekEngine


class LLMEngineFactory(Factory[LLMEngine]):
    def create(self, config: object) -> LLMEngine:
        if not isinstance(config, LLMConfig):
            config = LLMConfig()
        if not config.api_key:
            raise ValueError("LLMConfig.api_key is required")
        return DeepSeekEngine(config)

File: lynne-cpp/src/wheel/llm/llm_factory.cpp
#include "wheel/llm/llm_factory.h"
#include "wheel/llm/imp/deepseek_engine.h"

namespace lynne {
namespace wheel {

LLMEngine* LLMEngineFactory::create(const LLMConfig& config) const {
    return new DeepSeekEngine(config);
}

} // namespace wheel
} // namespace lynne
